skimage/morphology/_skeletonize.py

```diff
@@ -77,7 +77,7 @@ def skeletonize(image, *, method=None):
     """
 
     if image.ndim == 2 and (method is None or method == 'zhang'):
-        skeleton = skeletonize_2d(image)
+        skeleton = skeletonize_2d(image.astype(bool))
```

Comment:
I don't think this is true in this scenario. `image.astype(bool)` will do what it says in the docstring.

Reply:
(whereas I have no idea what `img_as_bool` will do!)

Reply:
Gonna trust you then, Juan :slightly_smiling_face:

Reply:
Sorry didn't have time to answer before the merge :confounded: 
`img_as_bool` is not what we want here because it ends up calling these lines

https://github.com/scikit-image/scikit-image/blob/5de98ebc253a4da8502cbb52021821bfec0a7a01/skimage/util/dtype.py#L266-L267=

which will binarise the image according to the median value of the input image dtype.
But your comment made me realize that I forgot to set the `copy` argument of `astype` to `False` :facepalm:
I have seen few improvements that can be made also in the Cython code. I will come with a new PR concerning these small improvements.
Thank you and @jni for your reviews :wink:

Reply:
Thank you for clarifying @rfezzani!